ngrinder-controller/src/main/java/org/ngrinder/agent/service/AgentPackageService.java

```diff
@@ -259,6 +243,20 @@ public boolean evaluate(Object object) {
 		}
 	}
 
+	private void copyShellFile(TarArchiveOutputStream tarOutputStream, String basePath, String type) throws IOException {
+		ResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+		Resource[] resources = resolver.getResources("classpath*:" + "ngrinder-sh/" + type + "/*");
+		InputStream shellFileIs;
+		byte[] shellFileBytes;
+		for (Resource resource : resources) {
+			shellFileIs = resource.getInputStream();
+			shellFileBytes = IOUtils.toByteArray(shellFileIs);
+			IOUtils.closeQuietly(shellFileIs);
+			addByteToTar(tarOutputStream, shellFileBytes, basePath + FilenameUtils.getName(resource.getFilename()),
+				shellFileBytes.length, EXEC);
```

Comment:
Isn't it better to use the following way? 
```
try (InputStream is = resource.getInputStream()) {
   ....
}
```

Reply:
TarArchiveOutputStream is not implements `java.lang.AutoCloseable` 
So, It can't use try-catch-resource.